src/app/tests/TestSceneTable.cpp

```diff
@@ -681,6 +722,77 @@ void TestHandlerFunctions(nlTestSuite * aSuite, void * aContext)
 
     memset(failBuffer, 0, fail_list.size());
     memset(buffer, 0, buff_span.size());
+
+    // Test Serialize Add of an attribute value that is greater than the mock attribute max (Max int32_t value)
+    OOPairs[0].attributeValue = 0x7FFFFFFF'FFFFFFFF;
+
+    // EFS to test caping of value once a variable above the mock attribute size is serialized
+    app::Clusters::ScenesManagement::Structs::ExtensionFieldSet::Type extensionFieldValueCapOut;
+    app::Clusters::ScenesManagement::Structs::ExtensionFieldSet::DecodableType extensionFieldValueCapIn;
+
+    extensionFieldValueCapOut.clusterID          = kOnOffClusterId;
+    extensionFieldValueCapOut.attributeValueList = OOPairs;
+
+    /// Setup of input EFS (by temporary using the output one)
+    writer.Init(buff_span);
+    NL_TEST_ASSERT(
+        aSuite, CHIP_NO_ERROR == app::DataModel::Encode(writer, TLV::AnonymousTag(), extensionFieldValueCapOut.attributeValueList));
+
+    reader.Init(buffer);
+    extensionFieldValueCapIn.clusterID = kOnOffClusterId;
+    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == reader.Next());
+    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == extensionFieldValueCapIn.attributeValueList.Decode(reader));
+
+    // Verify that the initial value is not capped
+    auto pair_iterator = extensionFieldValueCapIn.attributeValueList.begin();
+    pair_iterator.Next();
+    app::Clusters::ScenesManagement::Structs::AttributeValuePair::Type pair = pair_iterator.GetValue();
+    NL_TEST_ASSERT(aSuite, pair.attributeValue == OOPairs[0].attributeValue);
+
+    // Verify that we cap the value to the mock attribute size when serializing
+    NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == sHandler.SerializeAdd(kTestEndpoint1, extensionFieldValueCapIn, buff_span));
+    NL_TEST_ASSERT(aSuite,
+                   CHIP_NO_ERROR == sHandler.Deserialize(kTestEndpoint1, kOnOffClusterId, buff_span, extensionFieldValueCapOut));
+
+    // Verify that the output value is capped
+    NL_TEST_ASSERT(aSuite, extensionFieldValueCapOut.attributeValueList[0].attributeValue == 0x007FFFFF);
+
+    // Clear buffer
+    memset(buffer, 0, buff_span.size());
+
+    // Test Serialize Add of an attribute value that is smaller than the mock attribute min (-2)
+    OOPairs[0].attributeValue = 0xFFFFFFFF'FFFFFFFE;
```

Comment:
So I am confused.  I thought we were representing -2 as `0xFE` for a one-byte signed thing.  IS this one not one-byte?

Reply:
So the hypothetical attribute for this test is:
```
EmberAfAttributeMetadata mockmetadata = { .defaultValue  = EmberAfDefaultOrMinMaxAttributeValue(&minMaxValue),
                                          .attributeId   = 0,
                                          .size          = sizeof(int32_t),
                                          .attributeType = ZCL_INT24S_ATTRIBUTE_TYPE,
                                          .mask          = ATTRIBUTE_MASK_MIN_MAX }; // namespace
```
But we are simulating that -2 arrived on the wire in the for of an int64_t and it would be capped to the minimal value in the working type, hence giving -1 in a uint32_t.

As you have noted above, this convoluted example does not test every possible solution, and the spec is not exactly clear as to what the outcome should even be for that specific case, so more metadata and test will be added.